var/spack/repos/builtin/packages/pegtl/package.py

```diff
@@ -23,5 +23,24 @@ class Pegtl(CMakePackage):
     version('2.1.4', sha256='d990dccc07b4d9ba548326d11c5c5e34fa88b34fe113cb5377da03dda29f23f2')
     version('2.0.0', sha256='5aae0505077e051cae4d855c38049cc6cf71103a6cc8d0ddef01a576e8a60cc0')
 
+    variant("examples", default=True, description='Build example programs')
+    variant("tests", default=True, description='Build test programs')
```

Comment:
Are the tests and examples installed in the prefix directory?

Reply:
Yes, this effectively is "no-change" to the former behavior of pegtl when these variants are True.

Reply:
Ok, what I wanted to understand is if these variant affect the artifacts (executables, libraries etc.) that get deployed in the prefix. If yes, a variant would be well-suited otherwise we may want to seek other options.

Reply:
In fact looking closely it appears that `pegtl+examples+tests` (the former default) produces no additional artifacts _anywhere_ than `pegtl~examples~tests`.  The only difference is whether or not the compiler attempts to build the examples or test problems.  FWIW I'm trying to resolve issue #19313 with this PR.

Reply:
Can we drop the variants then, and compile tests conditionally when `self.run_tests` is true? There are many examples in the built-in repository if you search for `if self.run_tests:`

Reply:
Yep, working on it!

Reply:
New content in place & all required tests pass!